### PR TITLE
docs(compute-samples): init samples and test for delete protection

### PR DESCRIPTION
**Pull request by Sita04**
Fixes b/202503733

- [x] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [x] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] API's need to be enabled to test (tell us)
- [ ] Environment Variables need to be set (ask us to set them)
- [x] **Tests** pass:   `mvn clean verify` **required**
- [x] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] Please **merge** this PR for me once it is approved.

### CODE REVIEW

File: compute/cloud-client/src/main/java/compute/deleteprotection/SetDeleteProtection.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+          SetDeletionProtectionInstanceRequest.newBuilder()
+          .setProject(project)
+          .setZone(zone)
+          .setResource(instanceName)
+          .setDeletionProtection(deleteProtection)
+          .build();
+
+      instancesClient.setDeletionProtectionAsync(request).get();
+      // Double-check by retrieving the updated setting from the instance.
+      System.out.println("Updated Delete Protection setting.");
```
Comment (anguillanneuf) [inline compute/cloud-client/src/main/java/compute/deleteprotection/SetDeleteProtection.java:60]: Perhaps print the updated setting but not required.

**Review by anguillanneuf — APPROVED** [review]
A few nits but LGTM. Thank you.

File: compute/cloud-client/src/main/java/compute/deleteprotection/CreateInstanceDeleteProtection.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+  public static void main(String[] args)
+      throws IOException, ExecutionException, InterruptedException {
+    // TODO(developer): Replace these variables before running the sample.
+    // project: project ID or project number of the Cloud project you want to use.
+    // zone: name of the zone you want to use. For example: “us-west3-b”
+    // instanceName: name of the new virtual machine.
+    // deleteProtection: boolean value indicating if the new virtual machine should be
+    // protected against deletion or not.
+    String project = "your-project-id";
```
[the remark this answers is not in the document]
Comment (Sita04) [reply compute/cloud-client/src/main/java/compute/deleteprotection/CreateInstanceDeleteProtection.java:41]: Modified the variable init to:     
String projectId = "your-project-id-or-number"; 
Changed the var name to projectId to maintain consistency with other samples and languages.
Thanks!

File: compute/cloud-client/src/main/java/compute/deleteprotection/GetDeleteProtection.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import java.io.IOException;
+
+public class GetDeleteProtection {
+
+  public static void main(String[] args) throws IOException {
+    // TODO(developer): Replace these variables before running the sample.
+    // project: project ID or project number of the Cloud project you want to use.
+    // zone: name of the zone you want to use. For example: “us-west3-b”
+    // instanceName: name of the new virtual machine.
+    String project = "your-project-id";
```
[the remark this answers is not in the document]
Comment (Sita04) [reply compute/cloud-client/src/main/java/compute/deleteprotection/GetDeleteProtection.java:32]: Modified. Thanks.

**Pull request merged**